doc/mgr/orchestrator_modules.rst

```diff
@@ -150,6 +150,48 @@ specify a location when creating a stateless service.
 OSD services generally require a specific placement choice, as this
 will determine which storage devices are used.
 
+Error Handling
+--------------
+
+The main goal of error handling within orchestrator modules is to provide debug information to
+assist users when dealing with deployment errors.
+
+In detail, orchestrators need to explicitly deal with different kind of errors:
+
+1. No orchestrator configured
+
+.. autoclass:: NoOrchestrator
+
+2. An orchestrator doesn't implement a specific method.
+
+   For example, an Orchestrator doesn't support ``add_host``.
+
+   In this case, a ``NotImplementedError`` is raised.
+
+3. Missing features within implemented methods.
+
+   e.g. deploying iSCSI gateways is not supported
+
+.. autoclass:: MissingOrchestratorSupport
+
+4. Input validation errors
+
+   the ``orchestrator_cli`` module and other calling modules are supposed to
+   provide meaningful error messages.
+
+5. Errors when actually executing commands
+
+   The resulting Completion should contain an error string that assists in understanding the
+   problem.
+
+6. Invalid configuration in the orchestrator modules
+
+   This can be tackled similar to 5.
+
+
+All other errors should raise an exception that are then logged into the mgr log file.
```

Comment:
I suppose here that we should also return some sort of message similar (5)

Reply:
Hm. We may not be able to do so in all circumstances. This includes all bugs (e.g. including the _cli module before we create any Completions)

Reply:
I just meant that _some_ error message should be returned--it need not always contain any useful information, sometimes only instructions on how to go find more information about what went wrong.